### DISCUSSION TITLE
Running nextcloud-aio/collabora inside an unprivileged LXC/LXD container

### DISCUSSION THREAD

**Discussion by k4my4b** [nextcloud/all-in-one#2138] [Wiki]
[As of kernel version 5.19 it is possible to mount overlayfs atop idmapped layers.](https://www.phoronix.com/news/OverlayFS-IDMAPPED-Layers)
As such docker will choose overlay2 storage driver over other storage drivers, [however this driver is currently facing issues with mknod interception/emulation.](https://discuss.linuxcontainers.org/t/linux-6-1-lts-confirmed-but-no-fix-for-mknod-yet/16436/3) since docker supports both btrfs and zfs it is recommended that you use these storage drivers on a supported backing filesystem. 
You will need to force docker to use one of the above storage drivers as follows:
```bash
#/etc/docker/daemon.json

{
     "storage-driver":"btrfs"
}
```

**Comment by vicky-bs** [comment]
Trying to add "storage-driver":"zfs" since that is how my storage is configured does not allow docker deamon to start during boot I had to delete the daemon.json file to boot docker services
Setup - docker on privilege lxc container (Ubuntu)

**Reply by kri164** [reply]
How to set storage driver in case of podman?
